### PR TITLE
Support for tokenFile in client config

### LINKED ISSUE #1157: Add support for `tokenFile` field and automatic reload for the kubeconfig used in dashboard deployment
**What would you like to be added**:
- Support for `tokenFile` field in the `kubeconfig` that the dashboard deployment is using to connect to the Garden cluster. All other usages of kubeconfigs (where a kubeconfig can be passed from the outside) should not allow the `tokenFile` field.
- Automatic reload of the kubeconfig and the data referenced in it.

Ideally the dashboard should be able to start with a kubeconfig that looks like this:
```yaml
apiVersion: v1
kind: Config
clusters:
- cluster:
    certificate-authority-data: <CA-DATA>
    server: https://server
  name: garden
contexts:
- context:
    cluster: garden
    user: garden
  name: garden
current-context: garden
users:
- name: garden
  user:
    tokenFile: /var/run/secrets/projected/serviceaccount/token
```

A change in the referenced `tokenFile` should also be picked up.

**Why is this needed**:
This is needed in order to take advantage of [service account token volume projection](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#service-account-token-volume-projection). This will eliminate the use of static service account tokens that are never rotated in favor of short-lived tokens which will be renewed and maintained by the `kubelet`.

### PR DESCRIPTION
**Pull request by holgerkoser**
**What this PR does / why we need it**:
Add support for tokenFile field and automatic reload for the kubeconfig used in dashboard deployment.

**Which issue(s) this PR fixes**:
Fixes #1157

**Special notes for your reviewer**:
The kubernetes watches are not restarted when the `clientConfig` changes.  They will be restarted after 5-10 minutes with the new `clientConfig` values. 

**Release note**:

```feature operator
Automatically reload referenced data of the dashboards kubeconfig. This is necessary for the  Service Account Token Volume Projection feature.
```

### CODE REVIEW

File: packages/kube-config/lib/index.js [hunk 1]
```diff
@@ -55,159 +29,98 @@ function readKubeconfig (filename) {
...
-  try {
-    return getInCluster(env)
-  } catch (err) {
-    return fromKubeconfig(readKubeconfig())
-  }
+function testConfig () {
+  return Config.build({
+    server: 'https://kubernetes:6443'
+  }, {
+    token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwic3ViIjoic3lzdGVtOnNlcnZpY2VhY2NvdW50OmdhcmRlbjpkZWZhdWx0In0.-4rSuvvj5BStN6DwnmLAaRVbgpl5iCn2hG0pcqx0NPw'
```
[the remark this answers is not in the document]
Comment (holgerkoser) [reply packages/kube-config/lib/index.js:69]: Yes. I agree. Should I do it in this PR or in a separate one?

[the remark this answers is not in the document]
Comment (petersutter) [reply packages/kube-config/lib/index.js:69]: if it's too much effort, then let's do it in a separate PR

[the remark this answers is not in the document]
Comment (holgerkoser) [reply packages/kube-config/lib/index.js]: ✔️

**Review by petersutter — APPROVED** [review]
/lgtm

**Review by grolu — APPROVED** [review]
/lgtm
Please add a Release Note!

**Pull request merged**